### ISSUE TITLE
admission: improvements to KV admission

### ISSUE THREAD

**Issue by sumeerbhola** [cockroachdb/cockroach#65957]
Misc TODOs/limitations in KV admission control:
- [ ] Most work initiated from within KV bypasses admission control. This is not appropriate especially for background work like GC, backups etc. Fixing this mainly requires finding the places in the code where the low priority work originates and plumbing the appropriate `AdmissionHeader`. (Export, GC are now subject to admission control #71109, but there are TODOs regarding priority assignment)
- [x] There is no `KVLowPriWork` `WorkKind`, which means even low-priority KV work will get prioritized over user-facing SQL processing. (Edit: with the elastic classification, this is no longer true, since low-priority KV work goes through a different grant coordinator).
- [ ] The initial scan for `RangeFeedRequest` is not subject to admission control.
- [x] Read-only work is throttled when storage engines are overloaded due to writes. We need a way to distinguish work that includes writes, and make it pass through an additional `WorkQueue` that is hooked up to an `Granter` that only uses write overload as a resource signal.
- [x] When there are multiple stores, a single engine (for a particular store) being overloaded due to writes will cause work to be throttled for all stores.

Jira issue: CRDB-7813

**Comment by ajwerner** [comment]
Should we do something about `ExportRequest` and `AddSSTable`?

**Comment by sumeerbhola** [comment]
> Should we do something about ExportRequest and AddSSTable?

Yes. Since this is happening through `InternalServer.Batch`, it should be part of the first bullet.

**Comment by ajwerner** [comment]
Cool. For context, I'm raise this in the context of the recent issues involving interactions between the bespoke `ExportRequestLimiter` and its bad interactions with latching. I suspect we wouldn't have gotten ourselves into that pickle if we were thinking about admission control in a more centralized way.

https://github.com/cockroachdb/cockroach/blob/14bd70cb042bb304a5121d5ec7c0c8ac827410c7/pkg/kv/kvserver/store.go#L181-L193

**Comment by sumeerbhola** [comment]
>  interactions between the bespoke ExportRequestLimiter and its bad interactions with latching

Yep, I noticed the silliness with rate limiting after acquiring latches (when it got fixed). I don't think admission control would have helped -- since the CPU would not be saturated, it would allow low priority requests through. It is not aware that latches are a resource too, and that reducing latch contention due to low priority traffic will allow high priority traffic to get higher throughput. Modeling latches as a resource would be hard since we'd need to track contended spans.
